pkg/collector/check/id/id.go

```diff
@@ -11,15 +11,13 @@ import (
 	"fmt"
 	"hash/fnv"
 	"strings"
-
-	"github.com/DataDog/datadog-agent/comp/core/autodiscovery/integration"
```

Comment:
I couldn't import the this library in the autodiscovery component because of import cycle. Seeing that the only reason that we include it here is because `integration.Data#GetNameForInstance()`, I change the function signature to get the instance name `string` and instead of using `integration.Data` I used `[]byte`

Reply:
How does the cycle look like? I think ideally imports should reflect flow of data - autodiscovery provides data via subscriptions to multiple users, of which collector is just one example. It should not have any need to import any of the collector.

Reply:
So while I do agree with that statement, the autodiscovery component interface already import the `checkid` 
https://github.com/DataDog/datadog-agent/blob/c7eb6e5fd9102291c697a89e7ae4150ae93ae56a/comp/core/autodiscovery/component.go#L35

@david-ds do you have any thoughts?

Reply:
IIRC, but both `integration` package and `checkid` were created to break such cycles in the past, and basically contain a single type each. A cycle between two types sounds suspicious.

Maybe I missed something, but looking at the whole PR I don't see any import changes apart from this one. How did we get an import cycle if no new imports were added?

Reply:
Let me double check again, and see if there was something I had on a prior change